File: node_backend/src/models/response.js
const mongoose = require('mongoose')

const responseSchema = new mongoose.Schema({
    /*
    responses: [{
        response: {
            type: Number,
            required: true,
        }
    }],*/
    tScore: {
        type: Number,
        required: true,
    },
    catagory: {
        type: String,
    },
    problem: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Problem'
    },
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    submittedAt: { type: Date, default: Date.now }
})

const Response = mongoose.model('Response', responseSchema)

module.exports = Response

File: mentalheathprediction/lib/models/response.dart
class Response {
  final String id;
  final String problem;
  final String catagory;
  final String submittedAt;

  Response({this.id, this.problem, this.catagory, this.submittedAt});

  factory Response.fromJSON(Map<String, dynamic> json) {
    return Response(
        id: json['_id'],
        problem: json['problem']['name'],
        catagory: json['catagory'],
        submittedAt: json['submittedAt']);
  }
}
